adobe/frl-online-proxy
# Add upstream timeouts

## Summary
Add timeouts on the upstream calls to COPS.  This prevents hangs when running in forwarding mode, and prevents build-up of waiting requests in cache mode after the client has already closed their connection.


## Testing Steps
* Set up a debugging proxy like Charles.  Have it intercept calls to COPS and hold them on a breakpoint.
* Run the proxy in passthrough mode.
* If testing a debug build, set the environment variable FRL_PROXY_TIMEOUT to 10000 (for 10 seconds).
* Launch an app.

You will see the FRL proxy time out after 59 (or 10) seconds and return a failure to the client.

NOTE: This PR builds on the #25, so it should be merged after #25.  The proxy support makes it a lot easier to test.  Both PRs should be released at the same time.



Fixes #26
